### PR TITLE
feat(expr): split ParseURL operation into multiple URL extract operations

### PR DESCRIPTION
**Pull request by krzysztof-kwitt**

### CODE REVIEW

File: ibis/expr/types/strings.py [hunk 1]
```diff
@@ -673,6 +679,30 @@ def parse_url(
...
+    def host(self):
+        return ops.ExtractHost(self).to_expr()
+
+    def file(self):
+        return ops.ExtractFile(self).to_expr()
+
+    def path(self):
+        return ops.ExtractPath(self).to_expr()
+
+    def query(self, key: str | None = None,):
```
[the remark this answers is not in the document]
Comment (krzysztof-kwitt) [reply ibis/expr/types/strings.py:700]: Should we accept `StringValue` here also?

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/expr/types/strings.py:700]: Yeah, that should probably include `StringValue` even though not all backends probably support expressions there.

**Comment by krzysztof-kwitt** [comment]
@cpcloud I change type of this PR. It is not longer refactor only, because we make user-facing API changes.

**Comment by krzysztof-kwitt** [comment]
I haven't tested it for Snowflake, but it should be ready for merge.

File: ibis/backends/base/sql/registry/string.py [hunk 2]
```diff
@@ -65,15 +65,27 @@ def string_ilike(translator, op):
...
 
-    if key is None:
-        return f"parse_url({arg_formatted}, '{extract}')"
     else:
-        key_fmt = translator.translate(key)
-        return f"parse_url({arg_formatted}, '{extract}', {key_fmt})"
+
+        def _op(translator, op):
+            (arg,) = op.args
+            arg_formatted = translator.translate(arg)
```
Comment (cpcloud) [inline ibis/backends/base/sql/registry/string.py:85]:
```suggestion
            arg_formatted = translator.translate(op.arg)
```

File: ibis/backends/clickhouse/compiler/values.py [hunk 3]
```diff
@@ -1355,16 +1355,49 @@ def _last_value(op, **kw):
...
-@translate_val.register(ops.ParseURL)
-def _parse_url(op, **kw):
-    if (func := _PARSE_URL_FUNCS.get(extract := op.extract)) is None:
-        raise ValueError(
-            f"`{extract}` is not supported in the parse_url implementation for the "
-            "ClickHouse backend"
+@translate_val.register(ops.ExtractProtocol)
+def _extract_protocol(op, **kw):
+    arg = translate_val(op.arg, **kw)
+    return f"CAST(nullIf(protocol({arg}), '') AS Nullable(String))"
```
Comment (cpcloud) [inline ibis/backends/clickhouse/compiler/values.py:1361]: I think all the `CAST`s can be removed as `nullif` appears to have a `Nullable(String)` return type:

```
play-us :) explain header = 1 select nullif('a', '');

EXPLAIN header = 1
SELECT nullif('a', '')

Query id: d6de3f0f-2616-43ce-af61-6a1f1728a5a9

┌─explain─────────────────────────────────────┐
│ Expression ((Projection + Before ORDER BY)) │
│ Header: nullIf('a', '') Nullable(String)    │
│   ReadFromStorage (SystemOne)               │
│   Header: dummy UInt8                       │
└─────────────────────────────────────────────┘

4 rows in set. Elapsed: 0.001 sec.
```

**Review by cpcloud — CHANGES_REQUESTED** [review]
This looks great! A couple of small things.

File: ibis/backends/clickhouse/compiler/values.py (hunk 3 above)
Comment (mik-laj) [reply ibis/backends/clickhouse/compiler/values.py]: Done

**Review by cpcloud — APPROVED** [review]

**Comment by cpcloud** [comment]
Awesome, thank you!

**Pull request merged**